### PR TITLE
feat(billing): Add support for Cloud Control API search in Billing nav

### PR DESCRIPTION
**Pull request by velveteer**
image

image

### CODE REVIEW

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/773/gallery.html)

File: src/rxApp/rxApp.js [hunk 1]
```diff
@@ -427,13 +427,16 @@ angular.module('encore.ui.rxApp', ['encore.ui.rxAppRoutes', 'encore.ui.rxEnviron
...
 .directive('rxBillingSearch', function ($window) {
     return {
-        template: '<rx-app-search placeholder="Fetch account by transaction or auth ID..." submit="fetchAccounts">' +
-            '</rx-app-search>',
+        templateUrl: 'templates/rxBillingSearch.html',
         restrict: 'E',
         link: function (scope) {
+            scope.searchType = 'bsl';
+            scope.$watch('searchType', function () {
+                scope.placeholder = scope.searchType == 'bsl' ? 'Transaction or Auth ID' : 'Account or Contact Info';
```
Comment (chriscantu) [inline src/rxApp/rxApp.js]: @velveteer Should this be `..... scope.searchType === 'bsl'.......`?

**Comment by chriscantu** [comment]
@velveteer Other than my one question, LGTM

LGTM

**Comment by TheNando** [comment]
I concur with the usage of === instead of ==. After that change, LGTM.

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/773/gallery.html)

File: src/rxApp/rxApp.js [hunk 2]
```diff
@@ -427,13 +427,16 @@ angular.module('encore.ui.rxApp', ['encore.ui.rxAppRoutes', 'encore.ui.rxEnviron
...
         restrict: 'E',
         link: function (scope) {
+            scope.searchType = 'bsl';
+            scope.$watch('searchType', function () {
+                scope.placeholder = scope.searchType === 'bsl' ? 'Transaction or Auth ID' : 'Account or Contact Info';
+            });
             scope.fetchAccounts = function (searchValue) {
                 if (!_.isEmpty(searchValue)) {
-                    $window.location = '/billing/search/' + searchValue;
+                    $window.location = '/billing/search' + '?q=' + searchValue + '&type=' + scope.searchType;
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: Why not `$window.location = '/billing/search?q=' + ...`? Seems funny to perform `String + String` concatenation with static values.

Comment (velveteer) [inline src/rxApp/rxApp.js]: @CITguy Lol good catch.

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/773/gallery.html)

**Comment by CITguy** [comment]
LGTM

**Comment by chriscantu** [comment]
@parlarjb @velveteer LGTM - Even Better!

LGTM

File: src/rxApp/rxApp.js [hunk 3]
```diff
@@ -425,15 +425,24 @@ angular.module('encore.ui.rxApp', ['encore.ui.rxAppRoutes', 'encore.ui.rxEnviron
...
         restrict: 'E',
         link: function (scope) {
+            scope.searchType = 'bsl';
+            scope.$watch('searchType', function () {
+                scope.placeholder = scope.searchType === 'bsl' ? 'Transaction or Auth ID' : 'Account or Contact Info';
+            });
             scope.fetchAccounts = function (searchValue) {
                 if (!_.isEmpty(searchValue)) {
-                    $window.location = '/billing/search/' + searchValue;
+                    encoreRoutes.isActiveByKey('billing').then(function (isBilling) {
```
Comment (parlarjb) [inline src/rxApp/rxApp.js]: You forgot the explanatory comment :)

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/773/gallery.html)

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/773/gallery.html)

**Comment by comeatmebro** [comment]
[Wraith UI diff](http://eb4d4842ea70055541c8-b6a3ba129b5eb76af13f4d17e4c1a53b.r51.cf5.rackcdn.com/773/gallery.html)

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**